core/Archive/ArchiveInvalidator.php

```diff
@@ -120,17 +126,28 @@ public function forgetRememberedArchivedReportsToInvalidate($idSite, Date $date)
      * @param $idSites int[]
      * @param $dates Date[]
      * @param $period string
-     * @return array
+     * @param $segment Segment
+     * @param bool $cascadeDown
+     * @return InvalidationResultInfo
      * @throws \Exception
      */
-    public function markArchivesAsInvalidated(array $idSites, $dates, $period)
+    public function markArchivesAsInvalidated(array $idSites, array $dates, $period, Segment $segment = null, $cascadeDown = false)
     {
-        $dates = $this->removeDatesThatHaveBeenPurged($dates);
+        $invalidationInfo = new InvalidationResultInfo();
```

Comment:
Can the trailing `Info` from the class name be removed maybe to have `InvalidationResult`? `Info` is a very generic term and not really meaningful as everything is a "info". `InvalidationResult` is already very meaningful. It's not important though, would be only something tiny in case you get to work on it again anyway